core/src/main/scala/org/apache/spark/network/netty/NettyBlockRpcServer.scala

```diff
@@ -105,8 +105,14 @@ class NettyBlockRpcServer(
         val blockId = BlockId(uploadBlock.blockId)
         logDebug(s"Receiving replicated block $blockId with level ${level} " +
           s"from ${client.getSocketAddress}")
-        blockManager.putBlockData(blockId, data, level, classTag)
-        responseContext.onSuccess(ByteBuffer.allocate(0))
+        val blockStored = blockManager.putBlockData(blockId, data, level, classTag)
+        if (blockStored) {
+          responseContext.onSuccess(ByteBuffer.allocate(0))
+        } else {
+          val exception = new Exception(s"Upload block for $blockId failed. This mostly happens " +
```

Comment:
Instead of a new exception,
Can we use the exception object thrown at https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L1315?

Reply:
what specific exception class do you mean?

Reply:
I think @karuppayya is trying to say that we can throw specific exceptions from the "blockManager.putBlockData()" in order to pass exact failure message from server to client.

The "putBlockData" method of "BlockDataManager" interface has a boolean return type. So it can still return false in case it is not able to store a block. If this line throws exception, the same exception message will be passed to client as RPC failure - No change in that behavior.

This PR handles the scenario where it returns false. In that case also we should send RPC failure to client.